samtranslator/policy_templates_data/policy_templates.json

```diff
@@ -1729,6 +1729,41 @@
         ]
       }
     },
+    "StepFunctionsReadOnlyPolicy": {
+      "Description": "Gives readonly permission to a Step Functions",
+      "Parameters": {
+        "StateMachineName": {
+          "Description":"The name of the state machine to list its executions."
+        }
+      },
+      "Definition": {
+        "Statement": [
+          {
+            "Effect": "Allow",
+            "Action": [
+                "states:ListStateMachines",
+                "states:ListActivities",
+                "states:DescribeStateMachine",
+                "states:DescribeStateMachineForExecution",
+                "states:ListExecutions",
+                "states:DescribeExecution",
+                "states:GetExecutionHistory",
+                "states:DescribeActivity"
+            ],
+            "Resource": {
+              "Fn::Sub": [
+                "arn:${AWS::Partition}:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${stateMachineName}",
```

Comment:
Hi @ihalder1 , the arns for activities, executions and statemachines are different ([doc](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsstepfunctions.html#awsstepfunctions-resources-for-iam-policies)). Can you please add the resources and verify them?